File: src/main/java/org/manapart/enderports/ModBlocks.java
package org.manapart.enderports;

import net.minecraft.block.Block;
import net.minecraftforge.event.RegistryEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.registries.ForgeRegistries;

import static org.manapart.enderports.EnderPorts.teleporter;

@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD)
public class ModBlocks {
    @SubscribeEvent
    public static void registerBlocks(RegistryEvent.Register<Block> event) {
        if (!ForgeRegistries.BLOCKS.containsKey(teleporter.getRegistryName())) {
            ForgeRegistries.BLOCKS.register(teleporter);
        }
    }
}

File: src/main/java/org/manapart/enderports/ModBlocks.kt
package org.manapart.enderports

import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.registerObject

object ModBlocks {
    val teleporter = Teleporter()

    val REGISTRY = DeferredRegister.create(ForgeRegistries.BLOCKS, MODID)

    val TELEPORTER_BLOCK by REGISTRY.registerObject("teleporter") {
        teleporter
    }
}